utils/generate_sub_versions.py

```diff
@@ -0,0 +1,68 @@
+import random
+from difflib import SequenceMatcher
+
+
+DEFAULT_FREQUENCY = 1 / 30
+
+
+def generate_sub_versions(start_version, end_version, frequency=DEFAULT_FREQUENCY):
+    """
+    Generate intermediary versions between 2 versions of the same text, by applying successively 'insert' and 'delete'
+    modifications from the diff between start and end versions.
+
+    Args:
+        start_version (str): text version from which to start
+        end_version (str): text version to transform into
+        frequency (float): sampling ratio for sub-versions
+
+    Returns:
+        List[str]: list of sampled sub-versions, beginning by start version but not necessarily ending with end version
+
+    """
+    sub_versions = [start_version]
+    current_version = start_version
+    while current_version != end_version:
+        sequence_matcher = SequenceMatcher(None, current_version, end_version)
+
+        for tag, i1, i2, j1, j2 in sequence_matcher.get_opcodes():
+            if tag != 'equal':
+                text_before = current_version[:i1]
+                text_after = current_version[i2:]
+
+                if tag in ['delete', 'replace']:
+                    text_to_delete = current_version[i1:i2]
+                    for i in range(len(text_to_delete)):
+                        current_version = text_before + text_to_delete[i + 1:] + text_after
```

Comment:
If I understand correctly, you browse through `text_to_delete` from beginning to end. Meaning from left to right.
However when doing real-life `cmd+shift+z` isn't it the other way around?